hr_timesheet_holiday/models/hr_holidays.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Sunflower IT <http://sunflowerweb.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning as UserError
+
+
+class HrHolidays(models.Model):
+    """Update analytic lines on status change of Leave Request"""
+    _inherit = 'hr.holidays'
+
+    # Timesheet entry linked to this leave request
+    timesheet_ids = fields.One2many('hr.analytic.timesheet', 'leave_id',
+        'Timesheet entries')
+
+    @api.multi
+    def holidays_validate(self):
+        res = super(HrHolidays, self).holidays_validate()
+
+        # Postprocess Leave Types that have an analytic account configured
+        for leave in self:
+            account = leave.holiday_status_id.analytic_account_id
+
+            if account:
+
+                # Assert hours per working day
+                company = leave.employee_id.company_id
+                hours_per_day = company.timesheet_hours_per_day
+                if not hours_per_day:
+                    raise UserError(
+                        _("No hours per day defined for Company '%s'") %
+                        (company.name,))
+
+                # Assert user connected to employee
+                user = leave.employee_id.user_id
+                if not user:
+                    raise UserError(
+                        _("No user defined for Employee '%s'") %
+                        (leave.employee_id.name,))
+
+                # Add analytic line for the leave hours
+                hours = abs(leave.number_of_days) * hours_per_day
+                leave.sudo().timesheet_ids.unlink()
+                leave.sudo().write({'timesheet_ids': [(0, False, {
+                    'name': leave.name or leave.holiday_status_id.name,
+                    'date': leave.date_from,
```

Comment:
you should better iterate from `leave.date_from` to the end and add a timesheet line with the maximum per day. This looks better in the timesheet afterwards, and won't be bitten by some validations that might exist on timesheets